src/Sentry.OpenTelemetry/SentryPropagator.cs

```diff
@@ -0,0 +1,126 @@
+using Microsoft.Extensions.Primitives;
+using OpenTelemetry;
+using OpenTelemetry.Context.Propagation;
+
+namespace Sentry.OpenTelemetry;
+
+/// <summary>
+/// Sentry OpenTelemetry Propagator.
+/// Injects and extracts both <c>sentry-trace</c> and <c>baggage</c> headers from carriers.
+/// </summary>
+public class SentryPropagator : BaggagePropagator
+{
+    private readonly IHub _hub = SentrySdk.CurrentHub;
+
+    /// <summary>
+    /// <para>
+    ///     Creates a new SentryPropagator.
+    /// </para>
+    /// <para>
+    ///     You should register the propagator with the OpenTelemetry SDK when initializing your application.
+    /// </para>
+    /// <example>
+    ///     OpenTelemetry.Sdk.SetDefaultTextMapPropagator(new SentryPropagator());
+    /// </example>
+    /// </summary>
+    public SentryPropagator() : base()
+    {
+    }
+
+    internal SentryPropagator(IHub hub) : this()
+    {
+        _hub = hub;
+    }
+    /// <inheritdoc />
+    public override ISet<string> Fields => new HashSet<string>
+    {
+        SentryTraceHeader.HttpHeaderName,
+        BaggageHeader.HttpHeaderName
+    };
+
+    private static class OTelKeys
+    {
+        public const string SentryBaggageKey = "sentry.baggage";
+        public const string SentryTraceKey = "sentry.trace";
+    }
+
+    /// <inheritdoc />
+    public override PropagationContext Extract<T>(PropagationContext context, T carrier, Func<T, string, IEnumerable<string>> getter)
+    {
+        Debug.WriteLine("SentryPropagator.Extract");
+
+        var result = base.Extract(context, carrier, getter);
+        var baggage = result.Baggage; // The Otel .NET SDK takes care of baggage headers alread
+
+        Debug.WriteLine("Baggage");
+        foreach (var entry in baggage)
+        {
+            Debug.WriteLine(entry.ToString());
+        }
+
+        if (TryGetSentryTraceHeader(carrier, getter) is not {} sentryTraceHeader)
+        {
+            Debug.WriteLine("No SentryTraceHeader present in carrier");
+            return result;
+        }
+
+        Debug.WriteLine($"Extracted SentryTraceHeader from carrier: {sentryTraceHeader}");
+
+        var activityContext = new ActivityContext(
+            sentryTraceHeader.TraceId.AsActivityTraceId(),
+            sentryTraceHeader.SpanId.AsActivitySpanId(),
```

Comment:
This is possibly incorrect. The sentryTraceHeader.SpanId would be the Id of the upstream span right? Should we be using this to set the `SpanTracer.ParentSpanId` or`TransactionTracer.ParentSpanId` in the `SentrySpanProcessor` instead?

Reply:
It seems to be [what the Java code is doing](https://github.com/getsentry/sentry-java/blob/dc67004db625e4a09c3392d4ac3b7b38a9498d28/sentry-opentelemetry/sentry-opentelemetry-core/src/main/java/io/sentry/opentelemetry/SentrySpanProcessor.java#L103) though... 

```java
      final SpanId spanId = new SpanId(traceData.getSpanId());
```